public/app/plugins/datasource/prometheus/migrations/variableMigration.ts

```diff
@@ -0,0 +1,96 @@
+import { PromVariableQuery, PromVariableQueryType as QueryType } from '../types';
+
+const labelNamesRegex = /^label_names\(\)\s*$/;
+const labelValuesRegex = /^label_values\((?:(.+),\s*)?([a-zA-Z_][a-zA-Z0-9_]*)\)\s*$/;
+const metricNamesRegex = /^metrics\((.+)\)\s*$/;
+const queryResultRegex = /^query_result\((.+)\)\s*$/;
```

Comment:
Are those all the options that we can use? I remember we can use `scalar()` too.

Reply:
While scalar() is a function provided by Prometheus, these are function created by Grafana to list variables. The scalar() function can be used in the query_result() function like so, `query_result(scalar(<some prom query>))`